examples/goroutine_per_partition_consuming_manual_commit/main.go

```diff
@@ -0,0 +1,161 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"math/rand"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/twmb/franz-go/pkg/kgo"
+)
+
+type pconsumer struct {
+	quit chan struct{}
+	done chan struct{}
+	recs chan []*kgo.Record
+}
+
+var (
+	brokers = flag.String("b", "", "comma delimited brokers to consume from")
+	topic   = flag.String("t", "", "topic to consume")
+	group   = flag.String("g", "", "group to consume in")
+)
+
+func (pc *pconsumer) consume(topic string, partition int32, cl *kgo.Client) {
+	fmt.Printf("Starting consume for  t %s p %d\n", topic, partition)
+	for {
+		select {
+		case <-pc.quit:
+			pc.done <- struct{}{}
+			fmt.Printf("Closing consume for t %s p %d\n", topic, partition)
+			return
+		case recs := <-pc.recs:
+			// Mimick work to happen before committing records
+			time.Sleep(time.Duration(rand.Intn(150)+100) * time.Millisecond)
+			fmt.Printf("Some sort of work done, about to commit t %s p %d\n", topic, partition)
+			err := cl.CommitRecords(context.Background(), recs...)
```

Comment:
I wonder if you would prefer `MarkCommitRecords` here, and `AutoCommitMarks`? Perhaps I need another, `CommitMarks`.

This would allow more batching for your commits, vs. each commit here being blocking.

Not sure, what do you think? This is fine too.

Reply:
In our "real" usecase we dont commit that often, we rollup 10000 messages over a period of minutes so the commit per partition is called infrequently. For this example i like we stick to the commit per `[]*kgo.records` if your prefer to use the `markcommit` strategy for this example we will do that.